File: Convertor de Temperatura C++/src/kelvin.cpp
#include "kelvin.hpp"

void Kelvin::converte(float kelvin)
{
    cout << "Temperatura em Kelvin: " << setiosflags (ios::fixed) << setprecision (1) << kelvin << " K" << endl;
    cout << "Temperatura em Celsius: " << setiosflags (ios::fixed) << setprecision (1) << kelvin - 273 << " ºC" << endl;
    cout << "Temperatura em Fahrenheit: " << setiosflags (ios::fixed) << setprecision (1) << (kelvin - 273) * 9 / 5 + 32 << " ºF" << endl;
}

File: Convertor de Temperatura Java/src/dados/Kelvin.java
package dados;

public class Kelvin extends Termometro
{

	public Kelvin(double kelvin)
	{
		super(kelvin);
	}

	public void converte()
	{
		double celsius = getTemperatura() - 273;
		setCelsius(celsius);
		double fahrenheit = (getTemperatura() - 273) * 9 / 5 + 32;
		setFahrenheit(fahrenheit);
		double kelvin = getTemperatura();
		setKelvin(kelvin);
	}
	
}
